examples/src/bin/msgraph.rs

```diff
@@ -0,0 +1,114 @@
+//! Showcases how to define and use a nonstandard token type.
+//!
+//! Note: MSGraph requires you to set `client_id` and `client_secret` as extra
+//! parameters when performing the token exchange (see below).
+
+use oauth2::{Client, StandardToken, Url};
+
+use anyhow::{anyhow, Result};
+
+pub struct ConfigMS {
+    pub client_id: String,
+    pub client_secret: String,
+    pub tenant_domain: String,
+}
+
+pub fn config_from_args_ms(name: &str) -> Result<ConfigMS> {
+    let app = clap::App::new(name)
+        .about("Testing out OAuth 2.0 flows")
+        .arg(
+            clap::Arg::with_name("client-id")
+                .takes_value(true)
+                .long("client-id")
+                .help("Client ID to use."),
+        )
+        .arg(
+            clap::Arg::with_name("client-secret")
+                .takes_value(true)
+                .long("client-secret")
+                .help("Client Secret to use."),
+        )
+        .arg(
+            clap::Arg::with_name("tenant-domain")
+                .takes_value(true)
+                .long("tenant-domain")
+                .help("Tenant domain to use."),
+        );
+
+    let m = app.get_matches();
+
+    let client_id = m
+        .value_of("client-id")
+        .ok_or_else(|| anyhow!("missing: --client-id <argument>"))?
+        .to_string();
+    let client_secret = m
+        .value_of("client-secret")
+        .ok_or_else(|| anyhow!("missing: --client-secret <argument>"))?
+        .to_string();
+
+    let tenant_domain = m
+        .value_of("tenant-domain")
+        .ok_or_else(|| anyhow!("missing: --tenant-domain <argument>"))?
+        .to_string();
+
+    Ok(ConfigMS {
+        client_id,
+        client_secret,
+        tenant_domain,
+    })
+}
+
```

Comment:
Any chance you could remove this block entirely? I'm not sure what it's here for.

Reply:
These parameters are required for the example to work.
let config = config_from_args_ms("msgraph Example")?;

Reply:
What do you mean? This is just a comment and all of the information in it that is required for the example to run is either part of the example (the URLs) or provided as CLI arguments.

Reply:
OK... Sorry, the section shown for the question was not the comment...
I can delete it,

Reply:
Ah. I see. Sorry for the confusion!

Reply:
No worries... 
I just like to have comments of what expected...
But, your choice...